sentinel-ai-session/src/main/java/com/phonepe/sentinel/session/history/History.java

```diff
@@ -0,0 +1,30 @@
+package com.phonepe.sentinel.session.history;
+
+import com.fasterxml.jackson.annotation.JsonClassDescription;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.phonepe.sentinelai.core.agentmessages.AgentMessage;
+import lombok.Builder;
+import lombok.Value;
+
+import java.util.List;
+import java.util.Map;
+
+@Value
+@JsonClassDescription("memory to be used by AI Agents to learn new constructs and personalize conversations with users")
+@Builder
+public class History {
+
+    @JsonPropertyDescription("Session id for this session")
+    String sessionId;
+
+    @JsonPropertyDescription("List of list of messages exchanged by the agent and model in the conversation grouped by agent invocations")
+    List<List<AgentMessage>> messages;
```

Comment:
Keeping it this way so that people can remove the messages from a particular message in conversation if they want to keep the size of message controlled.

Reply:
This can be a map? there is a runId for every call to agent::execute iirc. Keep a trremap with key as {runId,firstmessageTime} sorted on firstMessageTime descending (latest first). I am guessing that in a lot cases only latest N messages (or maybe messages from Last N runs?)might be required.